src/service/viewport/viewport-impl.js

```diff
@@ -209,8 +209,14 @@ export class ViewportImpl {
     // https://github.com/ampproject/amphtml/issues/30838 for more details.
     // The solution is to make a "fake" scrolling API call.
     const isIframedIos = Services.platformFor(win).isIos() && isIframed(win);
-    if (isIframedIos) {
-      this.ampdoc.whenReady().then(() => win./*OK*/ scrollTo(-0.1, 0));
+    // Fix for https://github.com/ampproject/amphtml/issues/32165, we don't
+    // want to scroll if we're in a shadow doc.
+    const isShadowDoc =
+      this.ampdoc.getRootNode().nodeType !== /* DOCUMENT_NODE */ 9;
```

Comment:
Can we just check if `ampdoc.isSingleDoc()`?

Reply:
Done.

Reply:
Actually, this was a question for @dvoytenko. 😬

Reply:
Ah ok, will leave it as is unless @dvoytenko has a strong preference.

Reply:
Oh, yeah. `isSingleDoc` works great for this. Good idea.